File: frontend/src/api/video.js
// src/api/video.js — Video Consultation API helpers

const BASE = '/api/v1/video'

const authHeader = () => ({
  Authorization: `Bearer ${localStorage.getItem('access_token')}`,
  'Content-Type': 'application/json',
})

/** Doctor: create/start a Jitsi meeting room for an appointment */
export const createMeeting = (appointmentId) =>
  fetch(`${BASE}/create-meeting/${appointmentId}`, {
    method: 'POST',
    headers: authHeader(),
  }).then(r => r.json())

/** Doctor or Patient: get room name + join info */
export const getJoinInfo = (appointmentId) =>
  fetch(`${BASE}/join/${appointmentId}`, {
    headers: authHeader(),
  }).then(r => r.json())

/** Doctor: end the meeting and mark appointment completed */
export const endMeeting = (appointmentId) =>
  fetch(`${BASE}/end-meeting/${appointmentId}`, {
    method: 'POST',
    headers: authHeader(),
  }).then(r => r.json())

File: backend/app/api/v1/video.py
"""
Video Consultation — API Router
Embedded Jitsi Meet (no redirect); HTTP-safe for local development.
"""
from __future__ import annotations

from uuid import UUID

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from core.database import get_db
from dependencies.auth import get_current_user, get_doctor, get_any_user
from models.user import User
from services.video_service import VideoService

router = APIRouter(prefix="/video", tags=["Video Consultation"])


@router.post(
    "/create-meeting/{appointment_id}",
    summary="Doctor creates/starts a video meeting room",
)
async def create_meeting(
    appointment_id: UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_doctor),
):
    """
    Called by the doctor to open the Jitsi room.
    Sets `meeting_started_at` on the appointment.
    Returns room_name and display info for the Jitsi External API embed.
    """
    svc = VideoService(db)
    return await svc.create_meeting(appointment_id, current_user)


@router.get(
    "/join/{appointment_id}",
    summary="Get join info for a video meeting (doctor or patient)",
)
async def get_join_info(
    appointment_id: UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_any_user),
):
    """
    Returns room_name, display_name, is_host and meeting_started flag.
    Patients use `meeting_started` to know when to leave the waiting room.
    """
    svc = VideoService(db)
    return await svc.get_join_info(appointment_id, current_user)


@router.post(
    "/end-meeting/{appointment_id}",
    summary="Doctor ends the video meeting and completes appointment",
)
async def end_meeting(
    appointment_id: UUID,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_doctor),
):
    """
    Sets `meeting_ended_at`, transitions appointment status → completed.
    """
    svc = VideoService(db)
    return await svc.end_meeting(appointment_id, current_user)
